### PR TITLE
Div 8047 | Update API now supports updating provisional certificates

### PR DESCRIPTION
**Pull request by varadeth**

### CODE REVIEW

File: backend/certificate_api/src/services/certificate_service.js [hunk 1]
```diff
@@ -16,12 +30,15 @@ const sortCertificatesInUpdateTimeAscOrder = (certificates) => {
...
-    return sortedCerts[sortedCerts.length - 1];
-  }
+  const certificatesMap = getCertificatesByDose(certificates);
+  var keys = []
+  certificatesMap.forEach((value, key, certificatesMap) => {
+    keys.push(key)
+  });
+  keys.sort();
+  certificates = sortCertificatesInUpdateTimeAscOrder(certificatesMap.get(keys[keys.length-1]));
+  return certificates[certificates.length - 1];
```
Comment (dileepbapat) [inline backend/certificate_api/src/services/certificate_service.js:41]: Can you please add a test with data similar to records from db

Comment (varadeth) [reply backend/certificate_api/src/services/certificate_service.js:43]: Hi Dileep
I have added test case for this

File: backend/certificate_api/src/services/certificate_service.js [hunk 2]
```diff
@@ -5,23 +5,24 @@ const countries = require('i18n-iso-countries')
...
   }
 };
-
 const getLatestCertificate = (certificates) => {
-  if (certificates.length > 0) {
-    let sortedCerts = sortCertificatesInUpdateTimeAscOrder(certificates);
-    return sortedCerts[sortedCerts.length - 1];
-  }
+  certificates = sortCertificatesInUpdateTimeAscOrder(certificates);
+  return certificates[certificates.length - 1];
```
Comment (himeshr-egov) [inline backend/certificate_api/src/services/certificate_service.js:25]: if certificates is empty, then we would encounter exception here.

Comment (varadeth) [reply backend/certificate_api/src/services/certificate_service.js]: done

**Pull request merged**